pipelines/controller/src/main/java/org/openmrs/analytics/ApiController.java

```diff
@@ -15,41 +15,47 @@
  */
 package org.openmrs.analytics;
 
-import java.beans.PropertyVetoException;
-import java.io.IOException;
-import java.sql.SQLException;
 import org.apache.beam.sdk.metrics.MetricQueryResults;
 import org.openmrs.analytics.metrics.ProgressStats;
 import org.openmrs.analytics.metrics.Stats;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class ApiController {
+
   private static final Logger logger = LoggerFactory.getLogger(ApiController.class.getName());
 
   private static final String SUCCESS = "SUCCESS";
 
   @Autowired private PipelineManager pipelineManager;
 
   @PostMapping("/run")
-  public String runBatch(@RequestParam(name = "isFullRun", required = true) boolean isFullRun)
-      throws IOException, PropertyVetoException, SQLException {
+  public ResponseEntity<String> runBatch(
+      @RequestParam(name = "isFullRun", required = true) boolean isFullRun) {
     if (pipelineManager.isRunning()) {
-      throw new IllegalStateException("Another pipeline is running!");
+      return new ResponseEntity<>("Another pipeline is running.", HttpStatus.BAD_REQUEST);
```

Comment:
BAD_REQUEST doesn't seem to be a good candidate here.

Reply:
This error looks like happened mostly due to the client, as the the user is not supposed to make a request when another pipeline is running, hence I categorised it as a 400. Did you mean anything specifically in 4xx or should it be treated as a 5xx?

Reply:
I am suggesting 5xx. If client is not supposed to make a request then the control should be disabled on UI. This is clearly something wrong happening on the server side.

Reply:
Updated the Http status code to 500 in the latest commit.